fix: combine stdout and stderr in bash tool output

monotykamary opened the pull request:
## Summary
- Fix bash tool to combine stdout and stderr in output, ensuring commands like `bun test` that output to stderr are visible to users
- Stderr is still preserved separately in metadata for backward compatibility

## Test plan
- [x] Test with commands that output to stderr (e.g., `echo "test" >&2`)
- [x] Verify stderr is still available in metadata
- [x] Confirm existing functionality remains intact

Fixes #291

🤖 Generated with [opencode](https://opencode.ai)

monotykamary committed:
fix: combine stdout and stderr in bash tool output

Commands like `bun test` that output to stderr were not visible to users.
The bash tool now combines stdout and stderr in the main output while
preserving stderr separately in metadata for backward compatibility.

Fixes #291

🤖 Generated with [opencode](https://opencode.ai)

Co-Authored-By: opencode <noreply@opencode.ai>

monotykamary commented:
**Before:**
<img width="659" alt="image" src="https://github.com/user-attachments/assets/cd875305-7aef-40c2-912e-2fde7efb22d8" />


**After:**
<img width="678" alt="image" src="https://github.com/user-attachments/assets/7b8a4809-13e4-4c80-bdb5-cde61954e84d" />

nitsanavni reviewed (commented):
idea - also return process `exitCode` in metadata

monotykamary committed:
feat: add exitCode to bash tool metadata

Include process exit code in bash tool metadata to provide additional
context about command execution status. This helps users understand
whether commands succeeded or failed.

🤖 Generated with [opencode](https://opencode.ai)

Co-Authored-By: opencode <noreply@opencode.ai>

thdxr committed:
tweaks
